modules/js-and-css/optimize-oembeds/hooks.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Hook callbacks used for oEmbed Optimizer.
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+$oembed_lazy_load_scripts = false;
+
+/**
+ * Filter the oEmbed HTML.
+ *
+ * Add loading="lazy" to any iframe tags.
+ * Lazy load any script tags.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $html The oEmbed HTML.
+ * @return string
+ */
+function perflab_optimize_oembed_html( $html ) {
+	global $oembed_lazy_load_scripts;
+
+	$p = new WP_HTML_Tag_Processor( $html );
+
+	// Find the first iframe or script tag to act on.
+	while ( $p->next_tag() ) {
+		if ( 'IFRAME' === $p->get_tag() ) {
+			$loading_value = $p->get_attribute( 'loading' );
+			if ( empty( $loading_value ) ) {
+				$p->set_attribute( 'loading', 'lazy' );
+			}
+			return $p->get_updated_html();
+
+		} elseif ( 'SCRIPT' === $p->get_tag() && $p->get_attribute( 'src' ) ) {
+			$oembed_lazy_load_scripts = true;
+			$p->set_attribute( 'data-lazy-embed-src', $p->get_attribute( 'src' ) );
+			$p->set_attribute( 'src', '' );
+			return $p->get_updated_html();
+		}
+	}
+
+	return $html;
+}
+add_filter( 'embed_oembed_html', 'perflab_optimize_oembed_html', 10 );
+
+/**
+ * Add a script to the footer if there are lazy loaded embeds.
+ * Load the embed's scripts when they approach the viewport using an IntersectionObserver.
+ *
+ * @since n.e.x.t
+ */
+function perflab_optimize_oembed_lazy_load_scripts() {
+	global $oembed_lazy_load_scripts;
+
+	if ( ! $oembed_lazy_load_scripts ) {
+		return;
+	}
+	?>
+	<script type="module">
+        const lazyEmbedsScripts = document.querySelectorAll( 'script[data-lazy-embed-src]' );
+        const lazyEmbedScriptsByParents = new Map();
+
+        const lazyEmbedObserver = new IntersectionObserver( 
+            ( entries ) => {
+                for ( const entry of entries ) {
+                    if ( entry.isIntersecting ) {
+                        const lazyEmbedParent = entry.target;
+                        const lazyEmbedScript = lazyEmbedScriptsByParents.get( lazyEmbedParent );
+                        const embedScript = lazyEmbedScript.cloneNode();
+                        embedScript.src = lazyEmbedScript.dataset.lazyEmbedSrc;
```

Comment:
For some reason this is not working for me. I tested with a Tweet and a TikTok and neither of them had their scripts load. I was able to get them to load by creating a fresh `script` instead of cloning it:

```suggestion
                        const embedScript = document.createElement( 'script' );
                        for ( const attr of lazyEmbedScript.attributes ) {
                            if ( attr.nodeName === 'src' ) {
                                // Even though the src attribute is absent, the browser seems to presume it is present.
                                continue;
                            }

                            embedScript.setAttribute(
                                attr.nodeName === 'data-lazy-embed-src' ? 'src' : attr.nodeName,
                                attr.nodeValue
                            );
                        }
                        lazyEmbedScript.replaceWith( embedScript );
```

Reply:
Oh, the empty `src` is found because the PHP code has `$p->set_attribute( 'src', '' );` I think it can be updated to remove the attribute altogether?

Reply:
Just seeing this is included in my suggestion: https://github.com/WordPress/performance/pull/886#discussion_r1408292217